src/runtime/threads/thread_data.cpp

```diff
@@ -35,56 +35,17 @@ namespace hpx { namespace threads
     void intrusive_ptr_release(thread_data_base* p)
     {
         if (0 == --p->count_)
-            delete p;
-    }
-
-    ///////////////////////////////////////////////////////////////////////////
-    void* thread_data::operator new(std::size_t size, thread_pool& pool)
-    {
-        HPX_ASSERT(sizeof(thread_data) == size);
-
-        void *ret = reinterpret_cast<void*>(pool.allocate());
-        if (0 == ret)
-        {
-            HPX_THROW_EXCEPTION(out_of_memory,
-                "thread_data::operator new",
-                "could not allocate memory for thread_data");
-        }
-
-#ifdef HPX_DEBUG_THREAD_POOL
-        using namespace std;    // some systems have memset in namespace std
-        memset (ret, initial_value, sizeof(thread_data));
-#endif
-        return ret;
-    }
-
-    void thread_data::operator delete(void *p, std::size_t size)
-    {
-        HPX_ASSERT(sizeof(thread_data) == size);
-
-        if (0 != p)
         {
-            thread_data* pt = static_cast<thread_data*>(p);
-            thread_pool* pool = pt->pool_;
-            HPX_ASSERT(pool);
-
-#ifdef HPX_DEBUG_THREAD_POOL
-            using namespace std;    // some systems have memset in namespace std
-            memset (static_cast<void*>(pt), freed_value, sizeof(thread_data)); //-V598
-#endif
-            pool->deallocate(pt);
-        }
-    }
-
-    void thread_data::operator delete(void *p, thread_pool& pool)
-    {
-        if (0 != p)
-        {
-#ifdef HPX_DEBUG_THREAD_POOL
-            using namespace std;    // some systems have memset in namespace std
-            memset (p, freed_value, sizeof(thread_data));
-#endif
-            pool.deallocate(static_cast<thread_data*>(p));
+            if (thread_data* td = dynamic_cast<thread_data*>(p))
```

Comment:
@sithhell don't like it either. I thought about the approach with ```get_pool```. In this way ```is_created_from(pool*)``` also seems to be redundant. Another problem is that we have to check whether actual type is ```thread_data```, not ```stackless_thread_data```, because the latter doesn't actually use a pool for its creation/deletion.
Ideally, we should think about the interface of these three classes and specify, do we need creation from pool be defined in ```thread_data_base```?